lab-jonathan/test/test-stack.js

```diff
@@ -0,0 +1,81 @@
+let assert = require('chai').expect;
+let Stack = require('../lib/stack.js');
+
+describe('stack module', function(){
+  describe('.push()', function(){
+    it('should add a new node to the top of the stack', function(){
+      let st = new Stack();
+      st.push(25);
+      assert(st.top.val).to.equal(25);
+      assert(st.top.prev).to.equal(null);
+    });
+    it('should throw error when max size is exceeded', function(){
+      let st = new Stack(1);
+      let err = new ReferenceError('You overflowed your stack.');
+      let fn = function(){throw err;};
+      st.push(25);
+      assert(fn).to.throw(err);
```

Comment:
Sam thing as above in the Q. This is just a circular assignment, and isn't actually testing that an error is thrown. You do `st.push(25)`, which puts one thing on the stack, and you'll need to try and push again to actually throw the error.